core/src/main/java/io/questdb/griffin/engine/functions/str/MD5VarcharFunctionFactory.java

```diff
@@ -0,0 +1,32 @@
+/*******************************************************************************
+ *     ___                  _   ____  ____
+ *    / _ \ _   _  ___  ___| |_|  _ \| __ )
+ *   | | | | | | |/ _ \/ __| __| | | |  _ \
+ *   | |_| | |_| |  __/\__ \ |_| |_| | |_) |
+ *    \__\_\\__,_|\___||___/\__|____/|____/
+ *
+ *  Copyright (c) 2014-2019 Appsicle
+ *  Copyright (c) 2019-2024 QuestDB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.griffin.engine.functions.str;
+
+public class MD5VarcharFunctionFactory extends MD5StrFunctionFactory {
+    @Override
+    public String getSignature() {
+        return "md5(Ø)";
+    }
+}
```

Comment:
When yo unherit varchar (utf8) function from str (utf16) it will end up converting utf8 into utf16 to pass to Digest and then back to utf8 inside the digest. Instead it should implement the function differently

```JAVA
private static class MD5Func extends VarcharFunction implements UnaryFunction {

...

        @Override
        public Utf8Sequence getVarcharA(final Record rec) {
            final Utf8Sequence sequence = getArg().getVarcharA(rec);
            if (sequence == null) {
                return null;
            }
            sinkA.clear();
            this.hashFn.hash(sequence, sinkA);
            return sinkA;
        }

        @Override
        public Utf8Sequence getVarcharB(final Record rec) {
            ...
        }

``` 

Note that it uses `final Utf8Sequence sequence = getArg().getVarcharA(rec);` from the underlying parameter that returns Utf8 String.

Reply:
Thanks for your suggestion!

The proposed changes are implemented here: [dd80d90](https://github.com/questdb/questdb/pull/5156/commits/dd80d902e24453f8b927d57c36f258bab24ce4f6)

Originally I used `ToLowercaseFunctionFactory` and `ToLowercaseVarcharFunctionFactory` as bases for implementing the hash functions. I think they may have the same performance penalty of converting between encodings.